src/app/codegen-data-model-provider/CodegenDataModelProvider.h

```diff
@@ -80,6 +80,17 @@ class EnumeratorCommandFinder
     }
 };
 
+class DefaultProviderChangeListener : public DataModel::ProviderChangeListener
```

Comment:
lets place this in its own header. I also believe that `detail::app::detail::DefaultProviderChangeListener` does not make it obvious on what it is. This is not any default but rather maybe `EmberDirtyMarker` or `EmberProviderChangeListener` or `CodegenProviderChangeListener`

Thinking of it maybe eventually as we stabilize the API we can figure out a better name than `ProviderChangeListener` as that name makes me think of "a listener that gets notified when a provider chanbes" which is not at all what it is. Do you have any suggestions?

Something like `AttributeDirtyMarker` or `DataChangeNotifier` or something like that. Obviously not for this PR, but this PR made it obvious to me that the naming is not great.